### PR TITLE
Up to date NLog 5 with NLog 4.4.8

### PR DESCRIPTION
**Pull request by 304NotModified**

### CODE REVIEW

File: src/NLog/Targets/AsyncTaskTarget.cs [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+        /// Constructor
+        /// </summary>
+        protected AsyncTaskTarget()
+        {
+            TaskTimeoutSeconds = 150;
+
+            _taskStartNext = () => TaskStartNext(null);
+            _taskCompletion = TaskCompletion;
+            _taskCancelledToken = TaskCancelledToken;
+            _taskTimeoutTimer = new Timer(TaskTimeout);
```
Comment (304NotModified) [inline src/NLog/Targets/AsyncTaskTarget.cs]: @snakefoot 

could you help me with this one? The ctor is  not .net core compliant.

But more important, it looks like `_taskTimeoutTimer` is not used really in this class? (only dispose and change timer). `_taskTimeoutTimer` hasn't a callback method? Am I missing something?

Comment (snakefoot) [inline src/NLog/Targets/AsyncTaskTarget.cs]: The timer-callback-event is the TaskTimeout-method. Guess you could implement it like this:

`_taskTimeoutTimer = new Timer(TaskTimeout, null, Timeout.Infinite, Timeout.Infinite);`

Comment (snakefoot) [inline src/NLog/Targets/AsyncTaskTarget.cs]: Regarding the need for the Timer (to handle timeout handling), then this is the important line:

`_cancelTokenSource.Cancel();    // Notice how TaskCancelledToken auto recreates token`

Comment (304NotModified) [inline src/NLog/Targets/AsyncTaskTarget.cs]: but the timer never invoke any callback method?

Comment (snakefoot) [inline src/NLog/Targets/AsyncTaskTarget.cs]: Not sure I understand. TaskTimeout is the callback method, and the timer is started at this line:
`_taskTimeoutTimer.Change(TaskTimeoutSeconds * 1000, Timeout.Infinite);`
Anyway just remove the timer, then and see if the timeout test works :)

Comment (304NotModified) [inline src/NLog/Targets/AsyncTaskTarget.cs]: oops was reading "TaskTimeout" as the timeout in seconds. Thanks. Was using vscode as was to lazy to start vs

Comment (304NotModified) [inline src/NLog/Targets/AsyncTaskTarget.cs]: thanks!

**Comment by 304NotModified** [comment]
todo:

```
.NETCoreApp,Version=v1.0

/home/travis/build/NLog/NLog/tests/NLog.UnitTests/Layouts/JsonLayoutTests.cs(264,58): error CS0117: 'MethodBase' does not contain a definition for 'GetCurrentMethod'

```

File: src/NLog/Targets/AsyncTaskTarget.cs (hunk 1 above)
Comment (snakefoot) [inline src/NLog/Targets/AsyncTaskTarget.cs]: > Was using vscode as was to lazy to start vs

Explains the crazy indenting all over the NLog ver. 5 branch :)

Comment (304NotModified) [inline src/NLog/Targets/AsyncTaskTarget.cs]: I'm gonna blame auto merge ;)

Don't wanna format all to prevent merge conflicts.

**Comment by 304NotModified** [comment]
superseded by https://github.com/NLog/NLog/pull/2172

**Pull request closed without merging**